compiler/rustc_resolve/src/imports.rs

```diff
@@ -551,6 +552,33 @@ impl<'a, 'tcx> Resolver<'a, 'tcx> {
                     );
                 }
             });
+
+            for (_, resolution) in self.resolutions(module).borrow().iter() {
+                let resolution = resolution.borrow();
+                if let Some(binding) = resolution.binding
+                    && let Some(glob_binding) = resolution.shadowed_glob
+                    && let NameBinding { kind: NameBindingKind::Import { import, .. }, .. } = glob_binding
+                    && let Some(import_id) = import.id()
+                    && let import_def_id = self.local_def_id(import_id)
+                    && self.effective_visibilities.is_exported(import_def_id)
+                    && glob_binding.vis.is_public()
+                    && !binding.vis.is_public()
+                    && let Some(ns) = binding.res().ns()
+                {
+                    self.lint_buffer.buffer_lint_with_diagnostic(
+                        HIDDEN_GLOB_REEXPORTS,
+                        import_id,
```

Comment:
We need ID of `resolution.binding` here, not of `resolution.shadowed_glob`.

Reply:
Sorry how do you get `resolution.binding`'s NodeId? My bad attempt at it is:

```rust
let binding_id = match binding.kind {
    NameBindingKind::Res(res) => match res {
        def::Res::Def(_, def_id) => {
            if let Some(local_def_id) = def_id.as_local() {
                Some(self.def_id_to_node_id[local_def_id])
            } else {
                None
            }
        }
        def::Res::Local(node_id) => Some(node_id),
        _ => None,
    },
    NameBindingKind::Import { import, .. } => import.id(),
    _ => None,
};
```

which doesn't seem to be right at all? I tried searching through the docs and zulip but couldn't find relevant results.

Reply:
Many cases here are impossible, so something like
```rust
match binding.kind {
    Res(res) => self.def_id_to_node_id[res.def_id().expect_local()],
    Module(module) => self.def_id_to_node_id[module.def_id().expect_local()],
    NameBindingKind::Import { import, .. } => import.id(),
}
```
should work.